core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/kubernetes/KubernetesClient.scala

```diff
@@ -170,6 +170,15 @@ class KubernetesClient(
       }
     }.recoverWith {
       case e =>
+        Future {
+          kubeRestClient
```

Comment:
Please wrap this in a `blocking` block if the `kubeRestClient`'s actions are still all blocking. It will instruct the underlying Thread pool to create enough threads to deal with potentially blocked ones gracefully.

Reply:
> Please wrap this in a `blocking` block if the `kubeRestClient`'s actions are still all blocking. It will instruct the underlying Thread pool to create enough threads to deal with potentially blocked ones gracefully.
should you consider creating a failure for other reasons, such as a network timeout, then whether the deletion makes sense?

Reply:
should you consider creating a failure for other reasons, such as a network timeout, then whether the delete makes sense?

Reply:
> Please wrap this in a `blocking` block if the `kubeRestClient`'s actions are still all blocking. It will instruct the underlying Thread pool to create enough threads to deal with potentially blocked ones gracefully.

should you consider creating a failure for other reasons, such as network timeouts, whether delete still makes sense